### PR TITLE
Sudo init working group leads

### PR DESCRIPTION
**Pull request by mnaamani**
Modified the content directory schemas `initialize:dev` script to be configurable with two env variables to for sudo and lead key pairs so it can be re-used on production chain.

example:

```
SUDO_URI=//Alice LEAD_URI=//Alice yarn workspace @joystream/cd-schemas initialize:dev
```
I had previously also made a similar addition to the storage node cli commands to also initialize a lead storage provider on a production chain so I included it that work in this PR.

```
SUDO_URI=//Alice MEMBER_ID=1 MEMBER_URI=//Bob WORKER_URI=//Bob \
    yarn storage-cli sudo-create-sp
```

### CODE REVIEW

File: storage-node/packages/cli/src/commands/dev.ts [hunk 1]
```diff
@@ -142,4 +142,101 @@ const init = async (api: RuntimeApi): Promise<any> => {
...
+
+  api.workers.devBeginStorageOpeningReview(openingId)
+
+  debug(`Filling storage opening.`)
+  const providerId = await api.workers.devFillStorageOpening(openingId, applicationId)
+
+  debug(`Assigned storage provider id: ${providerId}`)
+}
+
+function getLeadOpeningInfo() {
```
Comment (Lezek123) [inline storage-node/packages/cli/src/commands/dev.ts]: It should be possible to use `GenericJoyStreamRoleSchema` type from `@joystream/hiring/schemas/role.schema.typings.ts` here for typesafety and then convert to json with `JSON.strigify`.

File: storage-node/packages/cli/src/commands/dev.ts [hunk 2]
```diff
@@ -142,4 +142,101 @@ const init = async (api: RuntimeApi): Promise<any> => {
   return check(api)
 }
 
-export { init, check, aliceKeyPair, roleKeyPair, developmentPort }
+// Using sudo create initial storage lead and worker with given keys taken from env variables.
+// Used to quickly setup a storage provider on a new chain before a council is ready.
+const makeMemberInitialLeadAndStorageProvider = async (api: RuntimeApi): Promise<any> => {
+  const sudoKey = getKeyFromAddressOrSuri(api, process.env.SUDO_URI)
```
Comment (Lezek123) [inline storage-node/packages/cli/src/commands/dev.ts:170]: When I don't provide the env variables I'm getting `Command Failed: TypeError: Cannot read property 'startsWith' of undefined` (It looks like it originates from `keyring.addFromUri` in `getKeyFromAddressOrSuri`)

File: storage-node/packages/cli/src/cli.ts [hunk 3]
```diff
@@ -72,6 +72,9 @@ const commands = {
   'dev-check': async (api) => {
     return dev.check(api)
   },
+  'sudo-create-sp': async (api) => {
```
Comment (Lezek123) [inline storage-node/packages/cli/src/cli.ts:75]: The `usage` const is not updated and still includes the old `vstore-init` command and no `sudo-create-sp`.

File: storage-node/packages/cli/src/commands/dev.ts [hunk 4]
```diff
@@ -142,4 +142,101 @@ const init = async (api: RuntimeApi): Promise<any> => {
...
+  const memberId = parseInt(process.env.MEMBER_ID)
+  const memberController = getKeyFromAddressOrSuri(api, process.env.MEMBER_URI).address
+  const leadAccount = memberController
+  const workerAccount = process.env.WORKER_ACCOUNT
```
Comment (Lezek123) [inline storage-node/packages/cli/src/commands/dev.ts]: At the beginning I tried using `WORKER_URI` as provided in PR description and this resulted in some arbitrary(?) role key (`5C4hrf...yUpnhM`).

When using `WORKER_ACCOUNT=5FHneW46xGXgs5mUiveU4sbTyGBzmstUspZC92UhjJM694ty` (Bob) the result is as expected.

**Review by Lezek123 — COMMENTED** [review]
Mostly works as expected, I added a few small comments regarding the storage CLI command.

File: storage-node/packages/cli/src/commands/dev.ts (hunk 1 above)
Comment (mnaamani) [reply storage-node/packages/cli/src/commands/dev.ts]: Thanks, will update to use this technique.

File: storage-node/packages/cli/src/commands/dev.ts (hunk 2 above)
Comment (mnaamani) [reply storage-node/packages/cli/src/commands/dev.ts:170]: Looks like I didn't do good input validation here. Will fix.

File: storage-node/packages/cli/src/cli.ts (hunk 3 above)
Comment (mnaamani) [reply storage-node/packages/cli/src/cli.ts:75]: will update.

File: storage-node/packages/cli/src/commands/dev.ts (hunk 4 above)
Comment (mnaamani) [reply storage-node/packages/cli/src/commands/dev.ts]: Yes I think it derived account from an empty string maybe. The description in the PR was incorrect. :)
I'll make some general improvements on input validation.

**Comment by mnaamani** [comment]
Applied suggested fixes and improved input validation. 
Renamed some env variables for the `sudo-create-sp` command:

```
SUDO_URI
MEMBER_ID
MEMBER_CONTROLLER_URI
STORAGE_WORKER_ADDRESS
```

**Review by Lezek123 — APPROVED** [review]
Works as expected.
The only small issue is that it fails with `SudoFailed` if the lead is already initialized, perhaps a simple check if lead is already set can be added before sending the sudo tx?

**Comment by mnaamani** [comment]
> Works as expected.
> The only small issue is that it fails with `SudoFailed` if the lead is already initialized, perhaps a simple check if lead is already set can be added before sending the sudo tx?

Yes I tested that as well, might as well add the check.

**Review by bwhm — APPROVED** [review]
lgtm!

**Pull request merged**